packages/gatsby-plugin-clerk/src/ssr/authenticateRequest.ts

```diff
@@ -0,0 +1,60 @@
+import { Clerk } from '@clerk/backend';
+import { isDevelopmentOrStaging } from '@clerk/backend-core/src/util/clerkApiKey';
+import { GetServerDataProps } from 'gatsby';
+import * as process from 'process';
+
+import { WithServerAuthOptions } from './types';
+import { parseCookies } from './utils';
+
+export async function authenticateRequest(context: GetServerDataProps, options: WithServerAuthOptions) {
+  const { loadSession, loadUser, loadOrganization, authorizedParties } = options;
+  const { headers } = context;
+  const cookies = parseCookies(headers);
+  const cookieToken = cookies['__session'];
+  const headerToken = (headers.get('authorization') as string)?.replace('Bearer ', '');
+  const apiKey = process.env.CLERK_API_KEY || '';
+  const frontendApi = process.env.GATSBY_CLERK_FRONTEND_API || '';
+  if (!apiKey || !frontendApi) {
+    // todo:
+    throw new Error('missing key');
```

Comment:
🔧 Do we need to do the check at this point, or should we leave Clerk SDK to raise it? So far, we've followed the latter approach.

Reply:
@nikosdouvlis Reminder☝️